src/query/expression/src/evaluator.rs

```diff
@@ -962,6 +1015,12 @@ impl<'a> Evaluator<'a> {
         match &args[0] {
             Value::Scalar(s) => match s {
                 Scalar::Array(c) => {
+                    if func_name == "array_fold" {
+                        let result = self.run_array_fold(c, &expr)?;
+                        let result_col = result.convert_to_full_column(expr.data_type(), 1);
+                        return Ok(Value::Scalar(Scalar::Array(result_col)));
```

Comment:
`array_fold` is an aggregation operation where the return value is no longer an array type